tests/lib/rules/require-default-props.js

```diff
@@ -952,6 +1105,32 @@ ruleTester.run('require-default-props', rule, {
       features: ['flow'],
       options: [{ forbidDefaultForRequired: true }],
     },
+    {
+      code: `
+        type Props = {
+          foo?: string,
+          bar?: string
+        }
+        function Hello({ foo = 'asdf', bar = 'qwer' }: Props) {
+          return <div>Hello {foo} and {bar}</div>;
+        }
+      `,
+      features: ['flow'],
```

Comment:
```suggestion
      features: ['types'],
```
i think this syntax works with TS too?